    <h1 id="firstHeading" class="firstHeading">std::coroutine_handle&lt;Promise&gt;::operator(), std::coroutine_handle&lt;Promise&gt;::resume</h1>            <table class="t-dcl-begin"> <tr class="t-dcl-h"> <td> Member of other specializations</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">void operator()() const;
void resume() const;</pre>
</td> <td> (1) </td> <td> <span class="t-mark-rev t-since-cxx20">(since C++20)</span> </td> </tr> <tr class="t-dcl-h"> <td> Member of specialization <code>std::coroutine_handle&lt;std::noop_coroutine_promise&gt;</code>
</td> <td></td> <td></td> </tr> <tr class="t-dcl t-since-cxx20"> <td> <pre data-language="cpp">constexpr void operator()() const noexcept;
constexpr void resume() const noexcept;</pre>
</td> <td> (2) </td> <td> <span class="t-mark-rev t-since-cxx20">(since C++20)</span> </td> </tr>  </table> <div class="t-li1">
<span class="t-li">1)</span> Resumes the execution of the coroutine to which <code>*this</code> refers, or does nothing if the coroutine is a no-op coroutine.</div> <div class="t-li1">
<span class="t-li">2)</span> Does nothing.</div> <p>The behavior is undefined if <code>*this</code> does not refer to suspended coroutine, or the coroutine is not a no-op coroutine and suspended at its final suspend point. A concurrent resumption of the coroutine may result in a data race.</p>
<p>Resumption of a coroutine on an execution agent other than the one on which it was suspended has implementation-defined behavior unless each execution agent either is a thread represented by <code><a href="../../thread/thread.html" title="cpp/thread/thread">std::thread</a></code> or <code><a href="../../thread/jthread.html" title="cpp/thread/jthread">std::jthread</a></code>, or is the thread executing <code>main</code>.</p>
<h3 id="Parameters"> Parameters</h3> <p>(none)</p>
<h3 id="Return_value"> Return value</h3> <p>(none)</p>
<h3 id="Exceptions"> Exceptions</h3> <p>If an exception is thrown from the execution of the coroutine, the exception is caught and <code>unhandled_exception</code> is called on the coroutine's promise object. If the call to <code>unhandled_exception</code> throws or rethrows an exception, that exception is propagated.</p>
<h3 id="Notes"> Notes</h3> <p>A coroutine that is resumed on a different execution agent should avoid relying on consistent thread identity throughout, such as holding a mutex object across a suspend point.</p>
<h3 id="Example"> Example</h3>  <h3 id="See_also"> See also</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="destroy.html" title="cpp/coroutine/coroutine handle/destroy"> <span class="t-lines"><span>destroy</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx20">(C++20)</span></span></span></div> </td> <td> destroys a coroutine <br> <span class="t-mark">(public member function)</span>  </td>
</tr> </table>           <div class="_attribution">
  <p class="_attribution-p">
    &copy; cppreference.com<br>Licensed under the Creative Commons Attribution-ShareAlike Unported License v3.0.<br>
    <a href="https://en.cppreference.com/w/cpp/coroutine/coroutine_handle/resume" class="_attribution-link">https://en.cppreference.com/w/cpp/coroutine/coroutine_handle/resume</a>
  </p>
</div>
